tests/compiler/std/math.release.wat

```diff
@@ -49905,7 +49905,7 @@
    call $~lib/builtins/abort
    unreachable
   end
-  f64.const -nan:0x8000000000000
+  f64.const nan:0x8000000000000
```

Comment:
Hmm, this some red flag. Why this happening?

Reply:
Maybe it's OK because debug.wat is not changed. But actually i don't know what happened.

Reply:
Could you sync with main and reinstall dependencies? May be you have some outdated binaryen

Reply:
Already sync with main. But onto solve it.

Reply:
I found location in tests:
```ts
assert(test_sqrt(-8.06684839057968084, NaN, 0.0, INVALID));
```
and other cases with negative input values.
After binaryen optimizations and inlining and DAE pass it should be equivalent to this:
```ts
check<f64>(-NaN, NaN, 0) &&
check<f64>(JSMath.sqrt(NaN), NaN, 0, INVALID));
```
But instead becomes:
```ts
check<f64>(+NaN, NaN, 0)
```

Reply:
Could you check in your node this line?:
```js
function signbit(x) {
    const F64 = new Float64Array(1);
    const U64 = new BigUint64Array(F64.buffer);
    return F64[0] = x, Boolean(U64[0] >> 63n);
}
console.log(signbit(Math.sqrt(-8.06684839057968084)));
```
Does it print `true`?

Reply:
Also does you rebuild test fixtures in `main` branch and check if std/math.ts is modified the same way? If not, then it is not a nondeterminism related to your env/platform and we should look for the problem in this PR.

Reply:
Yeah, it looks like it's non-determ on you env. I checkout this branch and rebuild std/math.ts. It reverts sign NaN back. May I ask what is your platform & env (os, cpu, node.js version)?

Reply:
It's definitely not a blocker for your PR, we can always port update the fixtures on our end. But I really wonder why this is happening on your machine. Maybe you should report the problem to v8 or binaryen

Reply:
I have try following env:
1. macos arm + node v16.9.1
2. macos arm + node v16.16.0

Reply:
Hmm, so it's M1. Interesting...